_studio/mfx_lib/scheduler/src/mfx_scheduler_core_ischeduler.cpp

```diff
@@ -328,7 +318,8 @@ mfxStatus mfxSchedulerCore::WaitForDependencyResolved(const void *pDependency)
 
         for (curIdx = 0; curIdx < m_numDependencies; curIdx += 1)
         {
-            if (m_pDependencyTable[curIdx].p == pDependency)
+            assert(m_pDependencyTable.size() > curIdx);
```

Comment:
I think you introduced too many excessive checks. You may use ```std::vector::at``` if you really fear of out-of-bounds access